rust/navigator/src/tests.rs

```diff
@@ -968,623 +3282,1424 @@ fn flow_test_1() {
-    assert!(
-        cut_real_json == expected_json,
-        "LongTap on Keys screen. Expected updated Keys screen, got:\n{}",
-        real_json
+    )
+    .unwrap()
+    .unwrap();
+    if let ScreenData::Keys { ref mut f } = expected_action.screen_data {
+        f.set[4].multiselect = false;
+        f.multiselect_count = "0".to_string();
+    }
+
+    assert_eq!(
+        action, expected_action,
+        "LongTap on Keys screen. Expected updated Keys screen.",
     );
 
     // Note: although multiselect count is 0, remain in multiselect mode
 
     do_action(
-        "LongTap",
+        Action::LongTap,
         "0120c394d410893cac63d993fa71eb8247e6af9a29cda467e836efec678b9f6b7f",
         "",
-    );
+    )
+    .unwrap();
     do_action(
-        "LongTap",
+        Action::LongTap,
         "012afba9278e30ccf6a6ceb3a8b6e336b70068f045c666f2e7f4f9cc5f47db8972",
         "",
-    );
+    )
+    .unwrap();
     do_action(
-        "LongTap",
+        Action::LongTap,
         "014e384fb30994d520094dce42086dbdd4977c11fb2f2cf9ca1c80056684934b08",
         "",
-    );
+    )
+    .unwrap();
     do_action(
-        "LongTap",
+        Action::LongTap,
         "01b606fc73f57f03cdb4c932d475ab426043e429cecc2ffff0d2672b0df8398c48",
         "",
-    );
+    )
+    .unwrap();
     do_action(
-        "LongTap",
+        Action::LongTap,
         "01e655361d12f3ccca5f128187cf3f5eea052be722746e392c8b498d0d18723470",
         "",
+    )
+    .unwrap();
+    // remove keys in multiselect mode
+    let action = do_action(Action::RemoveKey, "", "").unwrap().unwrap();
+
+    if let ScreenData::Keys { ref mut f } = expected_action.screen_data {
```

Comment:
Here, since `.remove()` is non-atomic, the number of the removed element looks somewhat confusing. Maybe at this point it already makes sense to redefine the `expected_action` variable.